### PR TITLE
Core: Fix Args test to not use different code path

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: We had special code for tests 🤮

### CODE REVIEW

File: lib/client-api/src/story_store.test.ts [hunk 1]
```diff
@@ -266,30 +273,38 @@ describe('preview.story_store', () => {
...
+            arg3: { complex: { object: ['changed'] } },
+            arg4: 'new',
+          },
         },
       });
       store.finishConfiguring();
 
       expect(store.getRawStory('a', '1').globalArgs).toEqual({
+        // You cannot remove a global arg in HMR currently
+        arg1: 'arg1',
```
Comment (tmeasday) [inline lib/client-api/src/story_store.test.ts:307]: Note this behaviour was *DIFFERENT* in the test using the wrong code path. Not quite sure why. 

I think? this is the correct behaviour though.

Comment (shilman) [reply lib/client-api/src/story_store.test.ts:307]: Doesn't make a lot of sense. The conditional codepath was just logging, right? I don't remember why I thought I needed to do that.

**Review by shilman — APPROVED** [review]
LGTM modulo my comment

File: lib/client-api/src/story_store.test.ts (hunk 1 above)
Comment (tmeasday) [reply lib/client-api/src/story_store.test.ts:307]: Nope the conditional code path was stopping you from setting global args per story

Comment (tmeasday) [reply lib/client-api/src/story_store.test.ts:307]: Still not sure why it changed the behaviour though

**Review by ndelangen — APPROVED** [review]

**Pull request merged**